src/parser/core/modules/AlwaysBeCastingTable.tsx

```diff
@@ -0,0 +1,304 @@
+		 * In this case, it's rounded up at 2 decimal places to give benefit of doubt
+		 */
+		const unroundedActionRecast: number
+			= actionGCDRecastTime === 0 ? gcdLength
+				: actionAttribute === undefined ? actionGCDRecastTime
+					: gcdLength / BASE_RECAST * actionGCDRecastTime
+		const actionRecast: number = Math.ceil(unroundedActionRecast * 100)/100
+		return actionRecast
+	}
+
+	private onComplete(event: Events['complete']) {
+		//finish up
+		this.checkAndSave(event.timestamp)
+
+		// Filter out periods where you got stunned, boss is untargetable, etc, or windows with negative durations
+		this.noCastWindows.history = this.noCastWindows.history.filter(windows => {
+			const duration = this.downtime.getDowntime(
+				windows.start,
+				windows.stop ?? windows.start,
+			)
+			return duration === 0
+				&& (
+					((windows.stop ?? windows.start) - windows.start > windows.recastTime + windows.casterTax + GCD_JITTER_OFFSET)
+					|| (windows.interruptedActions !== undefined && windows.interruptedActions?.length !== 0)
+				)
+		})
+	}
+
+	override output() {
+		if (this.noCastWindows.history.length === 0) { return <><Trans id="core.always-be-casting-table.no-violations">Nothing to note.</Trans> 😉</> }
+		return <><Table compact unstackable celled collapsing>
+			<Table.Header>
+				<Table.Row>
+					<Table.HeaderCell>
+						<Trans id="core.always-be-casting-table.timestamp-header">Timestamp</Trans><br/>
+						(<Trans id="core.always-be-casting-table.duration-header">Time between executed GCDs</Trans>)
+					</Table.HeaderCell>
+					<Table.HeaderCell><Trans id="core.always-be-casting-table.action-header">Relevant action(s)</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="core.always-be-casting-table.weaving-header"># of Weaves</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="core.always-be-casting-table.interrupted-header">Interrupted Actions</Trans>*</Table.HeaderCell>
+					<Table.HeaderCell>☠️</Table.HeaderCell>
+					<Table.HeaderCell><Trans id="core.always-be-casting-table.nothing-header">Doing nothing</Trans>**</Table.HeaderCell>
+				</Table.Row>
+			</Table.Header>
+			<Table.Body>
+				{this.noCastWindows.history.map(notCasting => {
+					return <Table.Row key={notCasting.start}>
+						<Table.Cell>
+							<Button
+								circular
+								compact
+								size="mini"
+								icon="time"
+								onClick={() => this.timeline.show(notCasting.start - this.parser.pull.timestamp, (notCasting.stop ?? notCasting.start) - this.parser.pull.timestamp)}
+							/><br/>
+							<span style={{marginRight: 5}}>{this.parser.formatEpochTimestamp(notCasting.start)}</span>
+							- <span style={{marginRight: 5}}>{this.parser.formatEpochTimestamp(notCasting.stop ?? notCasting.start)}</span>
+							<br/>
+							({this.parser.formatDuration((notCasting.stop ?? notCasting.start) - notCasting.start)})</Table.Cell>
+						<Table.Cell>
+							<Rotation events={[
```

Comment:
This is producing notably different results on a log I'm looking at to the main Weaving module: http://localhost:3000/fflogs/CJnbwdNzWmp9aT7K/58/90
- Appears to be showing a single weave at ~2 minutes, which weaving isn't complaining about. This might actually just be because they clipped their GCD to keep Starry perfectly on CD, but it's not clear from this output that's the case
- Appears to be complaining about a "triple weave" at ~8 minutes with only two weaved actions.
- Didn't catch the weave under Hyperphantasia at ~10 minutes

If this column is meant to be a replacement for the separate Weaving Issues table, then we're going to need to take a far closer look at how it's calculating those weaves and determining whether they're legitimate.

Reply:
100% definitely a lot to iron out
This module needs a bunch of love still and I want to incorporate your other feedback in the discord
The next draft should fix these issues (I have to incorporate your above feedback too and then I'll post the draft in discord again for feedback)
Thank you for this! I appreciate the feedback relating to pct because it's a lil new to me still

Reply:
So after looking at this, the reason the one around ~10 mins shows in weaving, but not in this is because there's an allowance of 0.05s for GCD_JITTER and they only violated gcd by 0.01s
If I lower the GCD_JITTER to 0, it shows, but then it's very punishing and a lot comes up under other issues
What do you think would be a good allowance if any?

(Additionally, I just noticed ignored actions for PCT and need to update for that, but that'll come in the next time I look at this)